### PR TITLE
Argparsed snapshot path

### PR DESCRIPTION
**Pull request by awilliamson**
Enables multiple executions as parameter, for hyperparameter search and
increase compatibility for scripting experiment runs.
As discussed with @hgaiser.

### CODE REVIEW

File: scripts/train.py [hunk 1]
```diff
@@ -192,6 +194,8 @@ def parse_args():
     parser.add_argument('--batch-size', help='Size of the batches.', default=1, type=int)
     parser.add_argument('--gpu', help='Id of the GPU to use (as reported by nvidia-smi).')
     parser.add_argument('--multi-gpu', help='Number of GPUs to use for parallel processing.', type=int, default=0)
+    parser.add_argument('--snapshot-path', help='Path to store snapshots of models during training (defaults to '
```
Comment (hgaiser) [inline scripts/train.py]: I'm fine with this being one line.

File: scripts/train.py [hunk 2]
```diff
@@ -218,7 +222,8 @@ def parse_args():
     print(model.summary())
 
     # create the callbacks
-    callbacks = create_callbacks(model, training_model, prediction_model, validation_generator, args.dataset_type)
+    callbacks = create_callbacks(model, training_model, prediction_model, validation_generator, args.dataset_type,
```
Comment (hgaiser) [inline scripts/train.py]: Same here.

File: scripts/train.py [hunk 3]
```diff
@@ -71,11 +71,13 @@ def create_models(num_classes, weights='imagenet', multi_gpu=0):
...
 
-def create_callbacks(model, training_model, prediction_model, validation_generator, dataset_type):
+def create_callbacks(model, training_model, prediction_model, validation_generator, dataset_type, snapshot_path):
     callbacks = []
 
     # save the prediction model
-    checkpoint = keras.callbacks.ModelCheckpoint(os.path.join('snapshots', 'resnet50_{dataset_type}_{{epoch:02d}}.h5'.format(dataset_type=dataset_type)), verbose=1)
+    checkpoint = keras.callbacks.ModelCheckpoint(os.path.join(snapshot_path,
+                                                              'resnet50_{dataset_type}_{{epoch:02d}}.h5'
+                                                              .format(dataset_type=dataset_type)), verbose=1)
```
Comment (hgaiser) [inline scripts/train.py]: I know this is what PEP8 suggests, but I like the alternative more. Ie. : 
```python
foo(
    bar=1,
    baz=2
)
```

Could you change it to that format? The rest of the code is formatted the same way.

**Comment by awilliamson** [comment]
@hgaiser Changes made. I took the liberty of liberally applying your style to the `ModelCheckpoint` as it could be cleaned further. Hopefully this meets your approval ( my editor did not want to cooperate on those PEP8 line constraints )

**Comment by hgaiser** [comment]
Looks good to me, thanks!

**Comment by yhenon-nextdroid** [comment]
Is this `./snapshots` safe for Windows users?

**Comment by yhenon** [comment]
Is this `./snapshots` safe for Windows users?

**Comment by de-vri-es** [comment]
Removing the `./` should have the same effect without any possible failure on Windows if the path is used properly. I do like the verbosity of `./snapshots`, but I suppose not breaking windows is more important. It might still work though... I'm never sure when forward slashes will actually break on windows and when not...

**Comment by awilliamson** [comment]
I'm unsure if `./` would be safe on windows. Was just a default action on my part. Isn't this parsed via some other functions rather than direct access to os? I imagine that wrapper will sanitise it? Does a windows user want to check this for us?

**Pull request merged**